checks/plugins.py

```diff
@@ -23,3 +23,63 @@ def checkImports(lines):
             append = "<br><br>Plugins affected:<ul>" + append + "</ul>"
             return [LEVEL_CRITICAL, "Outdated Plugins (" + str(len(conflicts)) + ")",
                     """Some plugins need to be manually updated, as they do not work with this version of OBS. Check our <a href="https://obsproject.com/kb/obs-studio-28-plugin-compatibility">Plugin Compatibility Guide</a> for known updates & download links.""" + append]
+
+
+def checkOperatingSystem(lines):
+    if 'mac' in lines[0]:
+        return "mac"
+    elif 'windows' in lines[0]:
+        return "windows"
+    elif 'linux' in lines[0]:
+        return "linux"
+
+
+def checkPluginList(lines):
+    if (getPluginLine(lines)):
+        windowsPlugins = ['win-wasapi', 'win-mf', 'win-dshow', 'win-capture', 'vlc-video', 'text-freetype2', 'rtmp-services', 'obs-x264', 'obs-websocket', 'obs-vst', 'obs-transitions', 'obs-text', 'obs-qsv11', 'obs-outputs', 'obs-filters', 'obs-ffmpeg', 'obs-browser', 'image-source', 'frontend-tools', 'decklink-output-ui', 'decklink-captions', 'coreaudio-encoder', 'win-decklink', 'decklink-ouput-ui', 'enc-amf']
+        macPlugins = ['vlc-video', 'text-freetype2', 'rtmp-services', 'obs-x264', 'obs-websocket', 'obs-vst', 'obs-transitions', 'obs-outputs', 'obs-filters', 'obs-ffmpeg', 'obs-browser', 'mac-virtualcam', 'mac-videotoolbox', 'mac-syphon', 'mac-capture', 'mac-avcapture', 'image-source', 'frontend-tools', 'decklink-output-ui', 'decklink-captions', 'coreaudio-encoder']
+        linuxPlugins = ['vlc-video', 'text-freetype2', 'rtmp-services', 'obs-x264', 'obs-websocket', 'obs-vst', 'obs-transitions', 'obs-outputs', 'obs-libfdk', 'obs-filters', 'obs-ffmpeg', 'obs-browser', 'linux-v4l2', 'linux-pulseaudio', 'linux-pipewire', 'linux-jack', 'linux-capture', 'linux-alsa', 'image-source', 'frontend-tools', 'decklink-output-ui', 'decklink-captions']
```

Comment:
Could we have a list `commonPlugins` here, so that we don't have to repeat the ones that are on all platforms?

Reply:
I thought about this as well but from what I could tell, it would just over complicate the code later on because of how the two arrays are being compared. There are still differences per operating system so it wouldn't eliminate the need to check the OS prior to comparing and I'd end up either doing multiple compares or combining arrays into new variables before checking. Bare in mind I haven't written anything in python prior to a week or so ago so I'm not claiming to be all-knowing, this is just what it seemed like to me.

Reply:
This was actually much easier than I had expected, PR has been updated.